manager.go

```diff
@@ -0,0 +1,150 @@
+package graphql
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/GannettDigital/graphql/gqlerrors"
+	"github.com/GannettDigital/graphql/language/ast"
+)
+
+const workerStartDelay = 250 * time.Microsecond
+
+// TODO add a count of workers and a way to report that
+// TODO add a way to reclaim workers that are idle
+
+// completeRequest contains the information needed to complete a field.
+// A completeRequest is passed to a resolveManager worker which processes the request.
+// This is only used when completing multiple items concurrently such as needed for a list
+type completeRequest struct {
+	index    int
+	response chan<- completeResponse
+
+	eCtx       *executionContext
```

Comment:
I haven't worked with executionContexts in here, would these ever get cancelled or timedout?

Reply:
Yes they inherit the context passed in as part of a request. I think in Do or the next function down it checks the context for cancel. Also of course the context is passed along so that DB calls and similar are checking it also.